src/movegen.cpp

```diff
@@ -140,22 +136,21 @@ namespace {
             if (pawnsNotOn7 & dcCandidates)
             {
                 Bitboard dc1 = shift<Up>(pawnsNotOn7 & dcCandidates) & emptySquares & ~file_bb(ksq);
-                Bitboard dc2 = shift<Up>(dc1 & TRank3BB) & emptySquares;
-
-                b1 |= dc1;
-                b2 |= dc2;
+                b |= dc1 | (shift<Up>(dc1 & TRank3BB) & emptySquares);
             }
         }
 
-        while (b1)
+        Bitboard singles = b & shift<Up>(pawnsNotOn7);
+        b ^= singles;
```

Comment:
You could just declare this as new variable `doubles` (`Bitboard doubles = b ^ singles;`) and use that below instead of `b`. The compiler should generate exactly the same code for both versions and I think the suggested one is easier to understand.

Reply:
Oh, I see now you had just that before the last commit... So please take this as a suggestion to reconsider :)